File: fix_quotes.js
const fs = require('fs');

const filePath = 'c:\\Users\\myafk\\Desktop\\antigravity2\\app.js';
let content = fs.readFileSync(filePath, 'utf8');

// Find lines within PARTNER_PRESET_COMMENTS and replace inner quotes
const lines = content.split('\n');
const newLines = [];

let inPresetSection = false;

for (let line of lines) {
    if (line.includes('PARTNER_PRESET_COMMENTS')) {
        inPresetSection = true;
    }
    if (inPresetSection && line.includes('// ====') && line.includes('State')) {
        inPresetSection = false;
    }

    if (inPresetSection && line.trim().startsWith('"')) {
        // Replace inner "word" patterns with 「word」
        // Pattern: the line is a string like "...\"word\"..."
        // We need to replace \" with 「 and then the closing \" with 」

        // Match pattern: \"something\" (escaped quotes inside the string)
        line = line.replace(/"([^"]+)"/g, function (match, p1, offset, string) {
            // Check if this is the outer quote (at start or end of trimmed string value)
            const trimmed = string.trim();
            if (offset === string.indexOf('"') || offset === string.lastIndexOf('"') - match.length + 1) {
                // This might be outer quote, but actually all quotes should be replaced except first and last
                return match;
            }
            return '「' + p1 + '」';
        });
    }

    newLines.push(line);
}

fs.writeFileSync(filePath, newLines.join('\n'), 'utf8');
console.log('Fixed!');

File: fix_quotes.py
import re

with open(r'c:\Users\myafk\Desktop\antigravity2\app.js', 'r', encoding='utf-8') as f:
    content = f.read()

lines = content.split('\n')
new_lines = []

in_preset_section = False
for line in lines:
    if 'PARTNER_PRESET_COMMENTS' in line:
        in_preset_section = True
    if in_preset_section and '// ====' in line and 'State' in line:
        in_preset_section = False
    
    if in_preset_section and line.strip().startswith('"'):
        # Replace inner double quotes with full-width quotes 「」
        # Pattern: inside a string, find "word" patterns that are inner quotes
        
        # Find the string content (between first " and last " or ",)
        match = re.match(r'^(\s*)(".*)(\"[,]?\s*)$', line)
        if match:
            indent = match.group(1)
            content_part = match.group(2)[1:]  # Remove opening quote
            end_part = match.group(3)
            
            # Replace inner "..." patterns with 「...」
            content_part = re.sub(r'"([^"]+)"', r'「\1」', content_part)
            
            line = indent + '"' + content_part + end_part
    
    new_lines.append(line)

with open(r'c:\Users\myafk\Desktop\antigravity2\app.js', 'w', encoding='utf-8') as f:
    f.write('\n'.join(new_lines))

print('Fixed!')
